torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -0,0 +1,181 @@
+    std::string to_backend_method_name = "_jit_to_" + backend_name;
+    m.def(
+        to_backend_method_name.c_str(),
+        [=](Module orig_module, py::dict method_compile_spec) {
+          // TODO: Validate method_compile_spec.
+
+          // Clone orig_module to make sure backend transformation is
+          // functional.
+          auto cloned_module = orig_module.clone();
+
+          // Represents of a Type of Dict[str, Any].
+          auto any_dict_ty =
+              DictType::create(StringType::get(), AnyType::get());
+
+          // Call preprocess.
+          auto* preprocess_fn = getCustomClassMethod(
+              c10::QualifiedName(backend_preprocess_prefix, backend_name));
+          TORCH_INTERNAL_ASSERT(preprocess_fn);
+          Stack stack;
+          push(
+              stack,
+              cloned_module._ivalue(),
+              toIValue(method_compile_spec, any_dict_ty).toGenericDict());
+          preprocess_fn->run(stack);
+          auto preprocessed_module = pop(stack);
+
+          // Create backend instance.
+          push(stack, IValue(preprocessed_module));
+          fn->run(stack);
+          auto backend = pop(stack).toCapsule();
+          TORCH_INTERNAL_ASSERT(backend);
+
+          // Generate LoweredModule.
+          Module loweredModule("torch.jit." + backend_name + "LoweredModule");
+
+          // Generate attributes.
+
+          // This is the original cloned and preprocessed module.
+          loweredModule.register_attribute(
+              "__processed_module", AnyType::get(), preprocessed_module);
+
+          // This is for the method_compile_spec passed in to to_<backend> or
+          // loaded from an exported model.
+          loweredModule.register_attribute(
+              "__method_compile_spec",
+              any_dict_ty,
+              toIValue(method_compile_spec, any_dict_ty).toGenericDict());
+
+          // This is a pointer to a backend instance that is used to access
+          // compile and execute functions via the prim::backend_compile and
+          // prim::backend_create primitives.
+          loweredModule.register_attribute(
+              "__backend", CapsuleType::get(), IValue::make_capsule(backend));
+
+          // This is the list of opaque backend handles returned by
+          // Backend::compile.
+          loweredModule.register_attribute(
+              "__handles",
+              any_dict_ty,
+              c10::impl::GenericDict(
+                  any_dict_ty->getKeyType(), any_dict_ty->getValueType()));
+
+          // Methods.
+          loweredModule.define(R"(
+            def __getstate__(self):
+                return self.__method_compile_spec, self.__processed_module
+            )");
+
+          loweredModule.define(
+              "\ndef __create_backend(self):\n\treturn ops.prim.backend_create(\"" +
+              backend_creator_name.qualifiedName() +
+              "\", self.__processed_module)\n");
```

Comment:
To make this easier to read, can we use [CodeTemplate](https://github.com/pytorch/pytorch/blob/dbe850af5b941c6f0ae1ed50e7d6b779db55f5af/torch/csrc/jit/frontend/code_template.h#L97)?

Reply:
Thanks, didn't know about that, will look into it.